astropy/coordinates/tests/test_spectral_coordinate.py

```diff
@@ -1074,11 +1074,29 @@ def test_spectralcoord_accuracy(specsys):
                 -row["obslon"], row["obslat"]
             ).get_itrs(obstime=row["obstime"])
 
-            with pytest.warns(AstropyUserWarning, match="No velocity defined on frame"):
+            with pytest.warns(AstropyUserWarning) as record:
                 sc_topo = SpectralCoord(
                     545 * u.nm, observer=observer, target=row["target"]
                 )
 
+            unique_warnings = {(wm.category, wm.message.args[0]) for wm in record}
+            expected = {
+                (
+                    NoDistanceWarning,
+                    (
+                        "Distance on coordinate object is dimensionless, "
+                        "an arbitrary distance value of 1000000.0 kpc will be set instead."
+                    ),
+                ),
+                (
+                    NoVelocityWarning,
+                    ("No velocity defined on frame, assuming (0., 0., 0.) km / s."),
+                ),
+            }
+            assert (
+                unique_warnings == expected
+            ), f"Got some unexpected warnings\n{unique_warnings - expected}"
```

Comment:
I don't think I have ever seen an assert statement where the manually specified error message is better than the one `pytest` creates. Could you post a comparison?

Reply:
Actually I can't, because I'm away and in reply-only mode until Wednesday. I wrote this in a hurry and didn't particularly think about how pytest would phrase the error if I let it. So feel free to patch this if you feel like it !

Reply:
I guess in pytest 8, now we have to catch 2 warnings instead of one? Let's get the jobs green first and then we can think about how to make it pretty.

Reply:
Yay, CI is green! @eerovaher , would you prefer the style of handling this like `astropy/wcs/tests/test_wcs.py` (near the diff you touched) or you thinking about something else?

Reply:
I don't see why we shouldn't use `pytest.warns()` to check for warnings here. Should I push a new commit?

Reply:
Sure. Exotic archs are a bit slow but so far things are green, so I think this PR is ready for any clean-up and possibly merged soon. Thanks!